### PR TITLE
FEATURE: web: build config.js on each boot

### PR DESCRIPTION
**Pull request by aaronkvanmeerten**
FEATURE: web: build system-config.js from tpl and env vars
FEATURE: web: include /config/custom-config.js if provided

### CODE REVIEW

**Comment by aaronkvanmeerten** [comment]
Work in progress, do not merge

**Comment by aaronkvanmeerten** [comment]
> Also I noticed you added the subdomain part. We need to fully bake that in the Docker setup, but that’s for another day :-)

I can also make an additional PR to the nginx config to enable it there, and a separate one to enable it in the prosody config.  This one simply addresses the needed pieces in config.js.  I'd like to keep that part in this PR as I'll need it to actual do any deployments using this method (or build a custom-config.js just for that until it does get merged, I suppose).

File: web/rootfs/defaults/settings-config.js
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:1]: doesn't the extra dash `-}}` break things?

File: web/rootfs/defaults/settings-config.js
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:6]:
```suggestion
{{ $CONFIG_RESOLUTION_WIDTH_MIN := .Env.CONFIG_RESOLUTION_WIDTH_MIN | default "320" -}}
```

File: web/rootfs/defaults/settings-config.js
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:16]: Let's not add the screen-sharing extension related options, since they no longer apply... we just need to find the time to axe the code.

File: web/rootfs/defaults/settings-config.js
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:62]:
```suggestion
Object.assign(config.testing, {});
```

I need to check if Duktape supports this though.

File: web/rootfs/defaults/settings-config.js
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:40]: shouldn't this go on the system ones?

File: web/rootfs/defaults/system-config.js
Comment (saghul) [inline web/rootfs/defaults/system-config.js:51]: Drop the fixme, it's never going to be fixed :-P

File: web/rootfs/defaults/settings-config.js
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:1]: We need to sort these options somehow (alphabetically?) so it doesn't become a mess, WDYT?

Comment (saghul) [reply web/rootfs/defaults/settings-config.js:1]: also, I'm not sure about having the defaults here. It means that the generated config will always be set and we have to keep it in sync with the internals. I'd rather pass it blindly I think, except for specific cases.

File: web/rootfs/etc/cont-init.d/10-config
Comment (saghul) [inline web/rootfs/etc/cont-init.d/10-config:78]: While we're here, same for interface_config?

File: web/rootfs/defaults/settings-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/settings-config.js:1]: Negatory, the dash simply removes whitespace after this statement, in this case the newline.  This keeps us from having a large number of newlines at the top of the file before the content.

File: web/rootfs/defaults/settings-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/settings-config.js]: Great, I've applied this everywhere

File: web/rootfs/defaults/settings-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/settings-config.js:40]: It can, it's not required, but it is more of a system setting, so I'll move it.

File: web/rootfs/etc/cont-init.d/10-config
Comment (aaronkvanmeerten) [reply web/rootfs/etc/cont-init.d/10-config:78]: I'm hesitant to tackle that part yet, since it's a whole separate bag of worms.  I'd prefer to make that a separate PR.

File: web/rootfs/defaults/settings-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/settings-config.js]: done

File: web/rootfs/defaults/system-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/system-config.js]: Done

File: web/rootfs/defaults/settings-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/settings-config.js]: Reverting this one:
Uncaught TypeError: Cannot convert undefined or null to object

File: web/rootfs/defaults/settings-config.js
Comment (aaronkvanmeerten) [reply web/rootfs/defaults/settings-config.js]: Removed defaults.  Also sorting the options alphabetically for the time being, until a better scheme comes along.

**Comment by aaronkvanmeerten** [comment]
Big updates, including config option renames to avoid redundancy, and increase consistency.  Needs another review please.

**Comment by ludovicm67** [comment]
Hi, thanks for your great work :+1: 

Can you also add support for some missing configuration keys?

It could be great if you could add support for Matomo (see here: https://github.com/jitsi/jitsi-meet/blob/6533071c4c91a5850c5545c81eea0df93ebf15aa/config.js#L404-L406) ; this will close #529.

Can you also add support for the `brandingDataUrl` key configuration (see: https://github.com/jitsi/jitsi-meet/blob/6533071c4c91a5850c5545c81eea0df93ebf15aa/config.js#L530)? This would help for #527.

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
I know we didn't cover all options, please feel free to send PRs for each individual one! 🚀

**Comment by nwvaras** [comment]
Is this working? It seems that if i add variables to the system-config or the settings-config it doesn't add to the config.js file. It seems like its been overrided by another file

File: web/rootfs/defaults/settings-config.js
Comment (chagai95) [reply web/rootfs/defaults/settings-config.js:16]: Is this working? I tried it and it didn't change the config.js @madmath03 @saghul

Comment (saghul) [reply web/rootfs/defaults/settings-config.js:16]: It should, yes.

**Pull request merged**
